tests/test_modeling_common.py

```diff
@@ -3010,7 +3011,7 @@ def check_device_map_is_respected(self, model, device_map):
             if param_device in ["cpu", "disk"]:
                 self.assertEqual(param.device, torch.device("meta"))
             else:
-                self.assertEqual(param.device, torch.device(param_device))
+                self.assertEqual(param.device, torch.device(f"{torch_device}:{param_device}"))
```

Comment:
it seems to me `f"{torch_device}:{param_device}"` would become `cuda:cuda` and I'm not sure `torch.device` will work with that. Do you try on cuda?

Reply:
yes, I tested it on CUDA and it works. The reason is that param_device are always intergers, but `param.device` not. 

The reason why param_device are always intergers is that `max_memory` only accepts integers as keys(see [code](https://github.com/huggingface/accelerate/blob/4ba436eccc1f6437503e66474d8ca86292f4acc1/src/accelerate/utils/modeling.py#L873) line), which makes the `device_map` returned by `infer_auto_device_map` has integer values as keys (see [code](https://github.com/huggingface/accelerate/blob/4ba436eccc1f6437503e66474d8ca86292f4acc1/src/accelerate/utils/modeling.py#L1151)), e.g. {'wte': 0, 'wpe': 0, 'drop': 0, 'ln_f': 1, 'h': 1}. This leads to the `hf_device_map` conataining intgers as keys (see [code](https://github.com/huggingface/accelerate/blob/4ba436eccc1f6437503e66474d8ca86292f4acc1/src/accelerate/big_modeling.py#L494)). 

And the reason why `param.device` is not integer is because accelerate is smart enough to detect the underlying hardware (see [code](https://github.com/huggingface/accelerate/blob/4ba436eccc1f6437503e66474d8ca86292f4acc1/src/accelerate/utils/modeling.py#L384)). 

The reason why the current code works is that torch.device(0) is equal to torch.device("cuda:0") automatically. But this is not good for other accelerators like XPU.

Reply:
These functions use this method: `test_disk_offload_bin`, `test_disk_offload_safetensors` , `test_cpu_offload`, `test_model_parallelism`. And all of them pass `max_memory` to `from_pretrained`. So this change won't make CUDA tests fail.

Reply:
This is a test example:
```bash
pytest tests/models/gpt2/test_modeling_gpt2.py::GPT2ModelTest::test_model_parallelism -rA
```

Reply:
thanks. will check!

Reply:
OK, thanks for explaining. I can see that `mps` could happen. But that is not the focus here.
cc @muellerzr here maybe?

Reply:
@muellerzr I am not 100% sure if we will ever have `mps` happens in `device_map`. In that case, this part of code seems have to be updated.

Reply:
Yeah, I think it is better if we do now even though we don't really test it on our CI. Maybe we can also add a comment on why we changed to `f"{torch_device}:{param_device}"` cc @faaany

Reply:
@SunMarc thanks for the answer! What should we do for `mps`? should I add an elif check like `elif param_device in ["mps] self.assertEqual(param.device, torch.device("mps"))"` ? 
Regarding the comment, can I add like this "# when loaded with device_map, `param_device` are integer values for cuda/xpu/npu/mlu"?

Reply:
Sounds good !